linux/LibSensors.c

```diff
@@ -92,13 +113,42 @@ int LibSensors_getCPUTemperatures(CPUData* cpus, int cpuCount) {
             if (r != 0)
                continue;
 
-            cpus[feature->number].temperature = temp;
-            tempCount++;
+            cpus[tempId].temperature = temp;
+            if (tempId > 0)
+               coreTempCount++;
          }
       }
    }
 
-   return tempCount;
+   const double packageTemp = cpus[0].temperature;
+
+   /* Only package temperature - copy to all cpus */
+   if (coreTempCount == 0 && !isnan(packageTemp)) {
+      for (unsigned int i = 1; i <= cpuCount; i++)
+         cpus[i].temperature = packageTemp;
+
+      return;
+   }
+
+   /* No package temperature - set to max core temperature */
```

Comment:
Missing the case with no package temperature and half of the core temperatures given.